imap-proto/src/lib.rs

```diff
@@ -7,18 +7,16 @@ extern crate assert_matches;
 
 #[macro_use]
 mod macros;
-mod parser;
 
-pub mod body;
-pub mod body_structure;
+// Public API
```

Comment:
Sorry, but this still has too much going on. Don't do reformatting in the same commit as changing the public API, please, because it's hard for me to review.